### PR TITLE
Final editor flow changes

### PR DESCRIPTION
**Pull request by scopsy**

### CODE REVIEW

**Comment by scopsy** [comment]
@ainouzgali There is quite a lot of changes here, suggest reviewing it per a commit here (Tried to make them concise and scoped to a single change)

File: apps/web/src/design-system/template-button/TemplateButton.tsx [hunk 1]
```diff
@@ -1,5 +1,6 @@
 import React from 'react';
 import { UnstyledButton, Group, Popover } from '@mantine/core';
+import styled from 'styled-components';
```
Comment (ainouzgali) [inline apps/web/src/design-system/template-button/TemplateButton.tsx:3]: Maybe its better to use emotion? To be consistent with mantine?

File: apps/web/src/components/templates/TemplatesSideBar.tsx [hunk 2]
```diff
@@ -133,3 +134,12 @@ export function TemplatesSideBar({
...
 }
+
+function getChannelErrors(channel: 'sms' | 'email' | 'inApp', errors: { [p: string]: string }) {
+  const keys = Object.keys(errors);
+  const channelErrors = keys.filter((key) => {
+    return key.includes(`${channel}Messages`);
+  });
+
+  return channelErrors.map((key) => errors[key]).toString();
+}
```
Comment (ainouzgali) [inline apps/web/src/components/templates/TemplatesSideBar.tsx:145]: Very cool :)

File: apps/web/src/design-system/button/Button.styles.ts [hunk 3]
```diff
@@ -8,17 +8,19 @@ const getLabelStyles = () => ({
...
   backgroundImage: theme.colorScheme === 'dark' ? getGradient(colors.B20) : getGradient(colors.B98),
 });
+
 const getFilledStyles = (theme) => ({
   border: 'transparent',
   boxShadow: theme.colorScheme === 'dark' ? shadows.dark : shadows.color,
 });
 
-const getOutlineStyles = (theme, disabled) => {
+const getOutlineStyles = (theme, disabled, variant?: 'outline' | 'filled') => {
```
Comment (ainouzgali) [inline apps/web/src/design-system/button/Button.styles.ts:21]: Why did you need the variant here? 🤔
I think it goes by class name of the variant

**Review by ainouzgali — APPROVED** [review]
Looks amazinggg  @scopsy🚀

File: apps/web/src/design-system/template-button/TemplateButton.tsx (hunk 1 above)
Comment (scopsy) [reply apps/web/src/design-system/template-button/TemplateButton.tsx:3]: Great catch! probably vscode autoimprot :(

Comment (scopsy) [reply apps/web/src/design-system/template-button/TemplateButton.tsx:3]: Thanks
